recipe-server/client/control_new/components/recipes/Listing.js

```diff
@@ -0,0 +1,185 @@
+  static propTypes = {
+    columns: PropTypes.object,
+    count: PropTypes.number,
+    fetchFilteredRecipesPage: PropTypes.func.isRequired,
+    getCurrentURL: PropTypes.func.isRequired,
+    ordering: PropTypes.string,
+    pageNumber: PropTypes.number,
+    push: PropTypes.func.isRequired,
+    recipes: PropTypes.object.isRequired,
+    searchText: PropTypes.string,
+    status: PropTypes.string,
+  };
+
+  static columnRenderers = {
+    name({ ordering }) {
+      return (
+        <Table.Column
+          title="Name"
+          dataIndex="name"
+          key="name"
+          render={Listing.renderLinkedText}
+          sortOrder={DataList.getSortOrder('name', ordering)}
+          sorter
+        />
+      );
+    },
+
+    action() {
+      return (
+        <Table.Column
+          title="Action"
+          dataIndex="action"
+          key="action"
+          render={Listing.renderLinkedText}
+        />
+      );
+    },
+
+    enabled({ status }) {
+      return (
+        <Table.Column
+          title="Enabled"
+          key="status"
+          render={(text, record) => <BooleanIcon value={record.enabled} />}
+          filters={[
+            { text: 'Enabled', value: 'enabled' },
+            { text: 'Disabled', value: 'disabled' },
+          ]}
+          filteredValue={status}
+          filterMultiple={false}
+        />
+      );
+    },
+
+    lastUpdated({ ordering }) {
+      return (
+        <Table.Column
+          title="Last Updated"
+          key="last_updated"
+          dataIndex="last_updated"
+          render={(text, record) => {
+            const lastUpdated = moment(record.last_updated);
+            return (
+              <Link href={`/recipes/${record.id}`} title={lastUpdated.format('LLLL')}>
+                {lastUpdated.fromNow()}
+              </Link>
+            );
+          }}
+          sortOrder={DataList.getSortOrder('last_updated', ordering)}
+          sorter
+        />
+      );
+    },
+  }
+
+  static renderLinkedText(text, record) {
+    return <Link href={`/recipes/${record.id}`}>{text}</Link>;
+  }
+
+  getFilters() {
+    const { ordering, searchText, status } = this.props;
+
+    const filters = {
+      text: searchText,
+      ordering,
+      status,
+    };
+
+    Object.keys(filters).forEach(key => {
+      if (filters[key] === undefined) {
+        delete filters[key];
+      }
+    });
+
+    return filters;
+  }
+
+  handleChangePage(page) {
+    const { getCurrentURL, push } = this.props;
+    push(getCurrentURL({ page }));
+  }
+
+  handleRowClick(record) {
+    const { push } = this.props;
+    push(`recipes/${record.id}`);
+  }
+
+  render() {
+    const { columns, count, ordering, pageNumber, recipes, status } = this.props;
+
+    return (
+      <div>
+        <QueryRecipes
+          pageNumber={pageNumber}
+          filters={this.getFilters()}
+        />
+        <LoadingOverlay>
+          <ListingActionBar />
```

Comment:
UX nit: Currently, this spinner disables the `New Recipe` button while loading:
<img width="790" alt="screen shot 2017-06-30 at 1 12 38 pm" src="https://user-images.githubusercontent.com/2767162/27752800-b38840fa-5d97-11e7-8193-cdb095b17356.png">

which is minor, but if a user is not interested in the table data, we're holding them up from doing what they want. I'd suggest just moving `ListingActionBar` out from under the `LoadingOverlay` for this reason.

--

Actually, better idea that requires more work: I think it'd look nice if we moved the `New Recipe` button up and in line with the breadcrumbs. That way, the only components 'attached' to the table are those that directly affect it.